doc/pce.sgml

```diff
@@ -0,0 +1,219 @@
+<toc>
+
+<!-- Begin the document -->
+
+<sect>Overview<p>
+
+This file contains an overview of the PCE runtime system as it comes
+with the cc65 C compiler. It describes the memory layout, PCE specific header
+files, available drivers, and any pitfalls specific to that platform.
+
+Please note that PCE specific functions are just mentioned here, they are
+described in detail in the separate <url url="funcref.html" name="function
+reference">. Even functions marked as "platform dependent" may be available on
+more than one platform. Please see the function reference for more
+information.
+
+
+<sect>Binary format<p>
+
+The standard binary output format generated by the linker for the PCE target
+is a cartridge image with no header. It is of course possible to change this
+behaviour by using a modified startup file and linker config.
+
+<sect>Memory layout<p>
+
+cc65 generated programs with the default setup run with the I/O area and a
+CHR bank enabled, which gives a usable memory range of &dollar;8000 - &dollar;FFF3.
+All boot ROM entry points may be called directly without additional code.
+
+Special locations:
+
+<descrip>
+  <tag/Text screen and Font/
+  The text screen is located at VRAM &dollar;0000,
+  the Font is located at VRAM &dollar;2000.
+
+  <tag/Stack/
+  The C runtime stack is located in system RAM at &dollar;3FFF and growing downwards.
+
+  <tag/BSS and Data/
+
+  The BSS (uninitialized variables) and Data (initialized variables) sections are
+  placed one after the other into system RAM at &dollar;2000.
+
+  <tag/Heap/
+  The C heap is located after the end of the Data section and grows towards the C
+  runtime stack.
+
+  <tag/Code/
+  The startup code is located at &dollar;E000 in the System/Hardware bank. Further
+  code can be placed in other ROM banks, this must be done manually however.
+
+</descrip><p>
+
+
+
+<sect>Platform specific header files<p>
+
+Programs containing PCE specific code may use the <tt/pce.h/ header file.
+
+
+<sect1>PCE specific functions<p>
+
+<itemize>
+<item>waitvblank</item>
+<item>get_tv</item> (since all PCE systems are NTSC, this always returns TV_NTSC)
+</itemize>
+
+
+
+<sect1>Hardware access<p>
+
+The following pseudo variables declared in the <tt/pce.inc/ include file do
+allow access to hardware located in the address space.
+
+<descrip>
+
+  <tag><tt/PSG/</tag>
+  The <tt/PSG/ defines allow access to the PSG chip (Programmable Sound Generator).
+
+  <tag><tt/VCE/</tag>
+  The <tt/VCE/ defines allow access to the VCE chip (Video Color Encoder).
+
+  <tag><tt/VDC/</tag>
+  The <tt/VDC/ defines allow access to the VDC chip (Video Display Controller).
+
+</descrip><p>
+
+
+
+<sect>Loadable drivers<p>
+
+All drivers must be statically linked because no file I/O is available.
+The names in the parentheses denote the symbols to be used for static linking of the drivers.
+
+
+<sect1>Graphics drivers<p>
+
+<descrip>
+
+No TGI graphics drivers are currently available for the PCE.
+
+</descrip><p>
```

Comment:
The `<descrip>` and `</descrip><p>` lines should be removed (there currently is nothing to describe there).